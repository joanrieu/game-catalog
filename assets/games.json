[
  {
    "name": "Rayman",
    "platform": "GBA",
    "cover": "//placehold.it/359x411"
  },
  {
    "name": "Rainbow Six",
    "platform": "PC",
    "cover": "//placehold.it/359x411"
  },
  {
    "name": "Just Dance",
    "platform": "Xbox",
    "cover": "//placehold.it/359x411"
  }
]
