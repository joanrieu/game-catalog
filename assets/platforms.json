["PC", "PS4", "X1"]
