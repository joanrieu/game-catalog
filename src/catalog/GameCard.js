import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Card = styled.section`
  display: flex;
  flex-direction: column;
  height: 100%;
  border-radius: 8px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  overflow: hidden;
`;

const Cover = styled.div`
  flex: 1;
  background-color: #c4c4c4;
  background-image: url(${props => props.cover});
  background-position: center;
  background-size: cover;
`;

const Description = styled.div`
  padding: 16px;
  background: white;
`;

const Name = styled.h2`
  font-weight: normal;
  font-size: 18px;
`;

const Platform = styled.p`
  margin-top: 8px;
  font-size: 14px;
`;

export default function GameCard({ name, platform, cover }) {
  return (
    <Card>
      <Cover cover={cover}> </Cover>
      <Description>
        <Name>{name}</Name>
        <Platform>{platform}</Platform>
      </Description>
    </Card>
  );
}

GameCard.propTypes = {
  name: PropTypes.string.isRequired,
  platform: PropTypes.string.isRequired,
  cover: PropTypes.string.isRequired
};
