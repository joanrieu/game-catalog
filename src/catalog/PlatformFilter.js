import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Wrapper = styled.nav`
  padding: 16px;
`;

const Platform = styled.button`
  border: none;
  background: transparent;
  font: inherit;
  color: ${props => (props.isSelected ? "#FE6050" : "inherit")}
  outline-color: #FE6050;
  text-transform: uppercase;

  &:not(:last-child) {
    margin-right: 28px;
  }
`;

export default function PlatformFilter({
  platforms,
  selectedPlatform,
  selectPlatform
}) {
  return (
    <Wrapper>
      <Platform onClick={() => selectPlatform()} isSelected={!selectedPlatform}>
        All
      </Platform>
      {platforms.map(platform => (
        <Platform
          key={platform}
          onClick={() => selectPlatform(platform)}
          isSelected={platform === selectedPlatform}
        >
          {platform}
        </Platform>
      ))}
    </Wrapper>
  );
}

PlatformFilter.propTypes = {
  platforms: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  selectedPlatform: PropTypes.string,
  selectPlatform: PropTypes.func.isRequired
};
