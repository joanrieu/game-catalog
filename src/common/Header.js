import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "react-router-dom";
import logoUrl from "../../assets/Video_game_controller_icon_designed_by_Maico_Amorim.svg";

const Wrapper = styled.header`
  background-color: #333333;
  color: white;
  display: flex;
  align-items: center;
`;

const Logo = styled.img`
  width: 32px;
  height: 32px;
  padding: 16px;
`;

const Title = styled.h1`
  font-weight: bold;
  font-size: 24px;
`;

export default function Header({ title }) {
  return (
    <Wrapper>
      <Link to="/">
        <Logo src={logoUrl} />
      </Link>
      <Title>{title}</Title>
    </Wrapper>
  );
}

Header.propTypes = {
  title: PropTypes.string.isRequired
};
