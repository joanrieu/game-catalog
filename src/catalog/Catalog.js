import React, { useState } from "react";
import PropTypes from "prop-types";
import Header from "../common/Header";
import PlatformFilter from "./PlatformFilter";
import GameCard from "./GameCard";

export default function Catalog({ platforms, games }) {
  const [selectedPlatform, selectPlatform] = useState();
  const filteredGames = selectedPlatform
    ? games.filter(game => game.platform === selectedPlatform)
    : games;
  return (
    <div>
      <Header title="Game catalog" />
      <PlatformFilter
        platforms={platforms}
        selectedPlatform={selectedPlatform}
        selectPlatform={selectPlatform}
      />
      <div>
        {filteredGames.map(game => (
          <GameCard key={game.name} {...game} />
        ))}
      </div>
    </div>
  );
}

Catalog.propTypes = {
  platforms: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  games: PropTypes.arrayOf(PropTypes.shape(GameCard.propTypes).isRequired)
    .isRequired
};
