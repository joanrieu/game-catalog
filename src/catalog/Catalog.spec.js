import React from "react";
import renderer from "react-test-renderer";
import Catalog from "./Catalog";
import Header from "../common/Header";
import PlatformFilter from "./PlatformFilter";
import GameCard from "./GameCard";
import platforms from "../../assets/platforms.json";
import games from "../../assets/games.json";

describe("Catalog", () => {
  it("has a header, a platform filter and a list of games", () => {
    const cmp = renderer.create(
      <Catalog platforms={platforms} games={games} />
    );
    expect(cmp).toMatchSnapshot();
  });
});
