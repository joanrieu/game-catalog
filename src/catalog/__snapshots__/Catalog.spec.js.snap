// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog has a header, a platform filter and a list of games 1`] = `
<div>
  <header
    className="sc-bdVaJa DnvYQ"
  >
    <img
      className="sc-bwzfXH ZQnxd"
      src="ASSET"
    />
    <h1
      className="sc-htpNat bqQPaO"
    >
      Game catalog
    </h1>
  </header>
  <nav
    className="sc-bxivhb hyktqA"
  >
    <button
      className="sc-ifAKCX lnbSfD"
      onClick={[Function]}
    >
      All
    </button>
    <button
      className="sc-ifAKCX vfORQ"
      onClick={[Function]}
    >
      PC
    </button>
    <button
      className="sc-ifAKCX vfORQ"
      onClick={[Function]}
    >
      PS4
    </button>
    <button
      className="sc-ifAKCX vfORQ"
      onClick={[Function]}
    >
      Xbox
    </button>
    <button
      className="sc-ifAKCX vfORQ"
      onClick={[Function]}
    >
      GBA
    </button>
  </nav>
  <div>
    <section
      className="sc-EHOje emDOoH"
    >
      <img
        className="sc-bZQynM cqXnZb"
        src="//placehold.it/359x411"
      />
      <div
        className="sc-gzVnrw jVtKtz"
      >
        <h2
          className="sc-htoDjs jJFWZZ"
        >
          Rayman
        </h2>
        <p
          className="sc-dnqmqq dpSZvM"
        >
          GBA
        </p>
      </div>
    </section>
    <section
      className="sc-EHOje emDOoH"
    >
      <img
        className="sc-bZQynM cqXnZb"
        src="//placehold.it/359x411"
      />
      <div
        className="sc-gzVnrw jVtKtz"
      >
        <h2
          className="sc-htoDjs jJFWZZ"
        >
          Rainbow Six
        </h2>
        <p
          className="sc-dnqmqq dpSZvM"
        >
          PC
        </p>
      </div>
    </section>
    <section
      className="sc-EHOje emDOoH"
    >
      <img
        className="sc-bZQynM cqXnZb"
        src="//placehold.it/359x411"
      />
      <div
        className="sc-gzVnrw jVtKtz"
      >
        <h2
          className="sc-htoDjs jJFWZZ"
        >
          Just Dance
        </h2>
        <p
          className="sc-dnqmqq dpSZvM"
        >
          Xbox
        </p>
      </div>
    </section>
  </div>
</div>
`;
